### PR TITLE
FEAT: wheelhouse action

### LINKED ISSUE #1: Implement fundamental workflows
🐞 **Problem**
There are common CI pipeline steps across different projects in PyAnsys. These steps are:

**Code style and tests**
- [x] Code style checking

**Documentation related**
- [x] Documentation style checking
- [x] Developers doc building and deployment
- [x] Stable doc building and deployment

**Software building related**
- [x] Creation of wheelhouse
- [x] Generation of wheel and sdist artifacts

*Software release related*
- [x] Private PyPI release publishing
- [x] Public PyPI release publishing
- [x] GitHub release

💡 **Solution**
Each one of previous steps should be a workflow itself. Reusable workflows should be seen like small modules.

Experience is that PyAnsys projects show small differences between them. These differences affect the workflows for the pipelines, making those kind of unique. 

We do not want to avoid opening a pull-request in a project, getting an error in the workflow, then coming to this repo to fix it and affecting lots of repositories.

### PR DESCRIPTION
**Pull request by jorgepiloto**
Fix #1 by adding the wheelhouse action.

### CODE REVIEW

File: build-wheelhouse/action.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: "Build the wheelhouse of a Python library"
+description: "Generate a Zip aritfacts for the wheelhouse of a Python library"
```
Comment (RobPasMue) [inline build-wheelhouse/action.yml:2]:
```suggestion
description: "Generate ZIP aritfacts for the wheelhouse of a Python library"
```

File: build-wheelhouse/action.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    - name: "Verify libray is properly installed and get its version number"
+      run: |
+        echo "library_version=$(python -c "from ${{ inputs.library-namespace }} import __version__; print(__version__)")" >> $GITHUB_ENV
+
+    - name: "Generate wheelhouse"
+      shell: bash
+      run: python -m pip wheel . -w wheelhouse
+
+    - name: "Zip wheelhouse"
```
Comment (RobPasMue) [inline build-wheelhouse/action.yml:51]: I thought we were going to proceed with ``cibuildwheel``

https://github.com/pyansys/pymapdl-reader/blob/8b0ace3f05e71b26fe632ab53538f2053bbee453/.github/workflows/testing-and-deployment.yml#L166-L167

File: doc/source/build-actions/index.rst [hunk 3]
```diff
@@ -28,6 +28,38 @@ Here is a code sample for using this action:
...
+             os: [ubuntu-latest, windows-latest]
+             python-version: ['3.7', '3.8', '3.9', '3.10']
+      steps:
+        - name: "Build a wheelhouse of the Python library"
+          uses: pyansys/actions/build-wheelhouse@main
+          with:
+            library-name: "<ansys-product-library>"
+            library-namespace: "<ansys.product.libray>"
+            operating-system: ${{ matrix.os }}
+            python-version: ${{ matrix.python-version }}
```
Comment (RobPasMue) [inline doc/source/build-actions/index.rst:60]: Why not just merge this action into a single call and provide the OS and Python versions as lists?

I guess that'd make it easier to use a plug-and-play stage of our workflow.

File: build-wheelhouse/action.yml (hunk 2 above)
Comment (jorgepiloto) [reply build-wheelhouse/action.yml:51]: Hmm... I may be confused. I thought the wheelhouse packaged all the dependencies so you do not need to install those from an index while the cibuildwheel generated wheels for desired platform. Does the cibuildwheel also include all the dependencies?

File: doc/source/build-actions/index.rst (hunk 3 above)
Comment (jorgepiloto) [reply doc/source/build-actions/index.rst:60]: It is a great idea! Let me check if reusable workflows support matrix strategy, not sure about this...

Comment (RobPasMue) [reply doc/source/build-actions/index.rst:60]: Let me know if it does! Sounds interesting =)

File: build-wheelhouse/action.yml (hunk 2 above)
Comment (RobPasMue) [reply build-wheelhouse/action.yml:51]: Ah yeah... that's true. But I thought maybe cibuildwheel was also able to pack all dependencies. Let me check...

File: doc/source/build-actions/index.rst (hunk 3 above)
Comment (jorgepiloto) [reply doc/source/build-actions/index.rst:60]: It is possible to call a composite workflow inside a matrix strategy. However, it is not possible to execute a matrix inside a composite workflow.

Comment (RobPasMue) [reply doc/source/build-actions/index.rst:60]: Damn... that's a shame =( thanks for looking into it anyway! =)

File: build-wheelhouse/action.yml (hunk 2 above)
Comment (RobPasMue) [reply build-wheelhouse/action.yml:51]: Yeah, looks like it only builds the specific wheel, not the entire wheelhouse. Resolving comment.

**Review by RobPasMue — APPROVED** [review]

**Pull request merged**